### PR TITLE
Support Ed25519ctx and Ed25519ph sign/verify algorithms

### PR DESCRIPTION
**Pull request by SparkiDev**

### CODE REVIEW

File: wolfssl/wolfcrypt/ed25519.h [hunk 1]
```diff
@@ -90,11 +96,35 @@ int wc_ed25519_make_public(ed25519_key* key, unsigned char* pubKey,
...
 int wc_ed25519_make_key(WC_RNG* rng, int keysize, ed25519_key* key);
 WOLFSSL_API
-int wc_ed25519_sign_msg(const byte* in, word32 inlen, byte* out,
-                        word32 *outlen, ed25519_key* key);
+int wc_ed25519_sign_msg(const byte* in, word32 inLen, byte* out,
+                        word32 *outLen, ed25519_key* key);
+WOLFSSL_API
+int wc_ed25519ctx_sign_msg(const byte* in, word32 inLen, byte* out,
+                           word32 *outLen, ed25519_key* key,
+                           const byte* context, byte contextLen);
```
Comment (toddouska) [inline wolfssl/wolfcrypt/ed25519.h]: Is byte the prober type for contextLen?  What's the max size now?

File: wolfcrypt/test/test.c [hunk 2]
```diff
@@ -18916,6 +18981,64 @@ int ed25519_test(void)
...
     }
 
+    outlen = sizeof(out);
+    XMEMSET(out, 0, sizeof(out));
+    
+    if (wc_ed25519_import_private_key(sKeyCtx1, ED25519_KEY_SIZE, pKeyCtx1,
+                                      sizeof(pKeyCtx1), &key) != 0)
+        return -9020;
+
+    if (wc_ed25519ctx_sign_msg(msgCtx1, sizeof(msgCtx1), out, &outlen, &key,
```
Comment (toddouska) [inline wolfcrypt/test/test.c]: Can we add a test for ctx_sign_msg with no context?

File: wolfcrypt/test/test.c [hunk 3]
```diff
@@ -18916,6 +18981,64 @@ int ed25519_test(void)
...
+#endif
+
+    outlen = sizeof(out);
+    XMEMSET(out, 0, sizeof(out));
+    
+    if (wc_ed25519_import_private_key(sKeyPh1, ED25519_KEY_SIZE, pKeyPh1,
+                                      sizeof(pKeyPh1), &key) != 0)
+        return -9025;
+
+    if (wc_ed25519ph_sign_msg(msgPh1, sizeof(msgPh1), out, &outlen, &key, NULL,
```
Comment (toddouska) [inline wolfcrypt/test/test.c]: Can we add a test for ph_sign_msg with a context?

Comment (toddouska) [reply wolfcrypt/test/test.c]: And both ph_sign_hash with and w/o a context?

File: wolfssl/wolfcrypt/ed25519.h (hunk 1 above)
Comment (SparkiDev) [reply wolfssl/wolfcrypt/ed25519.h]: The standard requires the context length to be between 0 and 255.
byte or uint8_t is the right type.

File: wolfcrypt/test/test.c (hunk 3 above)
Comment (SparkiDev) [reply wolfcrypt/test/test.c]: Added - no test vector in document
Added sign_hash and verify_hash tests.

File: wolfcrypt/test/test.c (hunk 2 above)
Comment (SparkiDev) [reply wolfcrypt/test/test.c]: Added - no test vector in document

**Pull request merged**